dspace-api/src/main/java/org/dspace/content/authority/SolrAuthority.java

```diff
@@ -135,14 +130,16 @@ public Choices getMatches(String text, int start, int limit, String locale,
                     }
                 }
 
-                if (externalResults && StringUtils.isNotBlank(text)) {
+                if (StringUtils.isNotBlank(text)) {
                     int sizeFromSolr = alreadyPresent.size();
-                    int maxExternalResults = limit <= 10 ? Math.max(limit - sizeFromSolr, 2) : Math
-                        .max(limit - 10 - sizeFromSolr, 2) + limit - 10;
+                    int maxExternalResults = sizeFromSolr < limit ? limit + 1 : sizeFromSolr + 1;
+                    // force an upper limit for external results
+                    if (maxExternalResults > 100) {
+                        maxExternalResults = 100;
```

Comment:
Setting the results to 100 might make requests quite slow. Because ORCID works like this:

* You do a query to retrieve a list of ORCID identifiers
* We send a single query for each ORCID identifier that we have to retrieve the details of that orcid

So in order to retrieve 100 records we need to send 101 requests to ORCID. I would recommend lowering this to 10.

Reply:
done